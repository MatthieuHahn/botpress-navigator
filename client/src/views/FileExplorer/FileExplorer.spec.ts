import { Store } from 'vuex-mock-store';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import FileExplorer from './FileExplorer.vue';
import {
  CONNECT_TO_SOCKET,
  GET_DIRECTORIES,
  SOCKET_MODULE
} from '@/store/actions/socket/socket.actions';
import { directoriesMock } from '@/tests/mocks/directories.mock';

const localVue = createLocalVue();
describe('FileExplorer', () => {
  let store;
  let wrapper;

  const getStore = () =>
    new Store({
      getters: {
        [`${SOCKET_MODULE}/directories`]: directoriesMock,
      },
      state: {
        [SOCKET_MODULE]: {

          actions: {
            [CONNECT_TO_SOCKET]: jest.fn(),
            [GET_DIRECTORIES]: jest.fn(),
          },
        },
      },
    });

  const whenComponentIsMounted = ({ $store = getStore() } = {}) => {
    store = $store;
    return shallowMount(FileExplorer, {
      mocks: {
        $store,
      },
      localVue,
      stubs: [
        'v-expansion-panels',
        'v-expansion-panel',
        'v-expansion-panel-header',
        'v-expansion-panel-content',
        'v-treeview',
      ],
    });
  };

  describe('when mounted', () => {
    beforeEach(() => {
      wrapper = whenComponentIsMounted();
    });

    it('should connect to websocket and get directories', () => {
      expect(store.dispatch)
        .toHaveBeenCalledWith(`${SOCKET_MODULE}/${CONNECT_TO_SOCKET}`);
      expect(store.dispatch)
        .toHaveBeenCalledWith(`${SOCKET_MODULE}/${GET_DIRECTORIES}`);
    });

    describe('and there are no directories', () => {
      beforeEach(() => {
        store.getters[`${SOCKET_MODULE}/directories`] = {};
      });

      it('should render an empty page', () => {
        expect(wrapper)
          .toMatchSnapshot();
      });
    });

    describe('and there are directories', () => {
      beforeEach(() => {
        store.getters[`${SOCKET_MODULE}/directories`] = directoriesMock;
      });

      it('should render the treeview for each directory', () => {
        expect(wrapper)
          .toMatchSnapshot();
      });
    });
  });
});
