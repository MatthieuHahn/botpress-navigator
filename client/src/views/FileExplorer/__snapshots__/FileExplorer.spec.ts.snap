// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileExplorer when mounted and there are directories should render the treeview for each directory 1`] = `
<v-expansion-panels-stub>
  <v-expansion-panel-stub>
    <v-expansion-panel-header-stub>
      <h2>directory1</h2>
    </v-expansion-panel-header-stub>
    <v-expansion-panel-content-stub>
      <v-treeview-stub items="[object Object],[object Object],[object Object]" activatable="" item-key="name" open-on-click=""></v-treeview-stub>
    </v-expansion-panel-content-stub>
  </v-expansion-panel-stub>
  <v-expansion-panel-stub>
    <v-expansion-panel-header-stub>
      <h2>directory2</h2>
    </v-expansion-panel-header-stub>
    <v-expansion-panel-content-stub>
      <v-treeview-stub items="[object Object],[object Object],[object Object]" activatable="" item-key="name" open-on-click=""></v-treeview-stub>
    </v-expansion-panel-content-stub>
  </v-expansion-panel-stub>
</v-expansion-panels-stub>
`;

exports[`FileExplorer when mounted and there are no directories should render an empty page 1`] = ``;
